source/viewmodel.ios.ts

```diff
@@ -272,11 +287,52 @@ class ImagePickerPH extends ImagePicker {
     }
 
     createPHImageThumb(target, asset: PHAsset): void {
-        PHImageManager.defaultManager().requestImageForAssetTargetSizeContentModeOptionsResultHandler(asset, this._thumbRequestSize, PHImageContentMode.PHImageContentModeAspectFill, this._thumbRequestOptions, function(target, uiImage, info) {
-            var imageSource = new image_source.ImageSource();
-            imageSource.setNativeSource(uiImage);
-            target.setThumb(imageSource);
-        }.bind(this, target));
+        PHImageManager.defaultManager().requestImageForAssetTargetSizeContentModeOptionsResultHandler(asset, this._thumbRequestSize, PHImageContentMode.PHImageContentModeAspectFill,
+            this._thumbRequestOptions, function (target, uiImage, info) {
+                var imageSource = new image_source.ImageSource();
+                imageSource.setNativeSource(uiImage);
+                target.setThumb(imageSource);
+            }.bind(this, target));
+    }
+
+    /**
+     * Creates a new ImageSource from the given image, using the given sizing options.
+     * @param image   The image asset that should be put into an ImageSource.
+     * @param options The options that should be used to create the ImageSource. 
+     */
+    createPHImage(image: PHAsset, options?: ImageOptions): Promise<image_source.ImageSource> {
+        return new Promise<image_source.ImageSource>((resolve, reject) => {
+            var size: CGSize = options ? CGSizeMake(options.maxWidth, options.maxHeight) : PHImageManagerMaximumSize;
+            var resizeMode = PHImageRequestOptions.alloc().init();
+
+            // TODO: Decide whether it is benefical to use PHImageRequestOptionsResizeModeFast
+            //       Accuracy vs Performance. It is probably best to expose these as iOS specific options.
+            resizeMode.resizeMode = PHImageRequestOptionsResizeMode.PHImageRequestOptionsResizeModeExact;
+            resizeMode.synchronous = false;
+
+            // TODO: provide the ability to change this setting.
+            //       Right now, it is needed to make sure that resolve is not called twice.
+            resizeMode.deliveryMode = PHImageRequestOptionsDeliveryMode.PHImageRequestOptionsDeliveryModeHighQualityFormat;
```

Comment:
PH can get thumb fast and download full photo from the cloud sometime later. If there is a need for this we will provide another API to obtain an observable that notifies each time new and better ImageSource is ready. As for this Promise based one - it is good.